bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java

```
@@ -304,7 +311,7 @@ public void processResponse(@Nullable Response response) {
                 reqStatus = sendRequest(reqStatus, config.deviceIp, COLOIT_URI_DEVSTATUS, Type.NON);
                 updatesRequested = true;
             }
-        } catch (JsonSyntaxException | IllegalArgumentException | NullPointerException e) {
+        } catch (Exception e) {
```

We are discussing to use IllegalArgumentException or RuntimeException all the time and now you catch all Exception generically. Also I thing explicitly point our that it could result in a JsonSyntaxException has a better quality. I would prefer JsonSyntaxException | RuntimeException

---

I noticed several calls to `org.eclipse.californium.core.coap`, i.e. a third party library outside of our control. If you are sure none of those exceptions are thrown by that library, we can remove the `catch` entirely. I checked and ensured that the binding doesn't throw any of those anymore.

---

If you parsed a malformed JSON to gson.fromJson() this will cause JsonSyntaxException. 
What do you think about JsonSyntaxException | RuntimeException?

---

Where do you see `gson.fromJson()` within this try/catch block?